src/Agent.Worker/Worker.cs

```diff
@@ -108,6 +108,17 @@ private void InitializeSecrets(JobRequestMessage message)
 
                 Trace.Verbose($"JobMessage: {channelMessage.Body}");
 
+                // Put system access token into variable.
```

Comment:
Consider moving this to job runner, preferably before the job-level execution context is created (I added variable expansion into the Variables ctor directly in my PR).

This Worker class is hard to test due to the parallel nature of the logic here. I suggest keeping this class limited to receiving the messages and managing core host-level things (setup secret masker, set thread culture).